client-rpc/src/rpc/wallet_rpc.rs

```diff
@@ -87,7 +95,63 @@ where
         }
     }
 
-    fn create(&self, request: WalletRequest) -> Result<String> {
+    fn create(&self, request: WalletRequest, kind: WalletKind) -> Result<String> {
+        match kind {
+            WalletKind::Basic => self.create_basic(request),
+            WalletKind::HD => self.create_hd(request),
+        }
+    }
+
+    fn create_basic(&self, request: WalletRequest) -> Result<String> {
+        if let Err(err) = self.client.new_wallet(&request.name, &request.passphrase) {
+            return Err(to_rpc_error(err));
+        }
+
+        self.client
+            .new_staking_address(&request.name, &request.passphrase)
+            .map_err(to_rpc_error)?;
+        self.client
+            .new_transfer_address(&request.name, &request.passphrase)
+            .map_err(to_rpc_error)?;
+        Ok(request.name)
+    }
+
+    fn create_hd(&self, request: WalletRequest) -> Result<String> {
+        let mnemonics = self.client.new_mnemonics().map_err(to_rpc_error)?;
+        let mnemonics_phrase = SecUtf8::from(mnemonics.to_string());
+
+        // make seed for hd-wallet
+        if let Err(err) =
+            self.client
+                .new_hdwallet(&request.name, &request.passphrase, &mnemonics_phrase)
+        {
+            return Err(to_rpc_error(err));
+        }
+        // make basic wallet
+        // only generation is different with basic wallet
+        if let Err(err) = self.client.new_wallet(&request.name, &request.passphrase) {
+            return Err(to_rpc_error(err));
+        }
+
+        self.client
+            .new_staking_address(&request.name, &request.passphrase)
+            .map_err(to_rpc_error)?;
+        self.client
+            .new_transfer_address(&request.name, &request.passphrase)
+            .map_err(to_rpc_error)?;
+        Ok(mnemonics.to_string())
+    }
+
+    fn restore(&self, request: WalletRequest, mnemonics: String) -> Result<String> {
+        let mnemonics_phrase = SecUtf8::from(mnemonics.to_string());
```

Comment:
```
    fn restore(&self, request: WalletRequest, mnemonics_phrase: SecUtf8) -> Result<String> {
```

Could `mnemonics_phrase` be passed directly as an argument instead of being converted here?